src/swipeable.js

```diff
@@ -265,13 +297,19 @@ class Swipeable extends Component {
       return;
     }
 
+    let nextTarget = this.state.direction === Swipeable.LEFT ? ++currentChildIndex : --currentChildIndex;
+
+    if (nextTarget < 0 ) {
+      nextTarget = 0;
+    } else if (nextTarget === this.childXCenterPosList.length) {
+      nextTarget = this.childXCenterPosList.length - 1;
+    }
+
     if (
       this.getDragVelocity() > this.props.flickSensitivity &&
       Math.abs(distance) <= this.content.childNodes[0].offsetWidth
```

Comment:
`// If velocity of swipe is greater than the specified flick sensitivity and the drag distance is lesser or equal to the first child width. A flick gesture has been initiated.`